### PR TITLE
make use of Stream API

### PR DESCRIPTION
**Pull request by grimreaper**
- hasControl() only needs a .contains call instead of a .get() call
- simplify some logic by using the streams API
- reduce visibility for things which aren't used outside of PC (makes it easier to spot errors in the future)

### CODE REVIEW

File: code/src/java/pcgen/core/PlayerCharacter.java [hunk 1]
```diff
@@ -10085,6 +10057,6 @@ public String getControl(String string)
 
 	public boolean hasControl(String string)
 	{
-		return controller.get(ObjectKey.getKeyFor(String.class, "*" + string)) != null;
+		return controller.containsKey(ObjectKey.getKeyFor(String.class, "*" + string));
```
Comment (thpr) [inline code/src/java/pcgen/core/PlayerCharacter.java]: IF the value can be set to null, these are not equivalent.  Have you checked for that case?

File: code/src/java/pcgen/core/PlayerCharacter.java [hunk 2]
```diff
@@ -8109,14 +8092,7 @@ private boolean isCrossClassSkill(Skill sk, PCClass pcc)
...
 	{
-		for (PCClass cl : getClassSet())
-		{
-			if (isCrossClassSkill(sk, cl))
-			{
-				return true;
-			}
-		}
-		return false;
+		return getClassSet().parallelStream().anyMatch(v -> isCrossClassSkill(sk, v));
```
Comment (thpr) [inline code/src/java/pcgen/core/PlayerCharacter.java]: thought we weren't doing parallel?

File: code/src/java/pcgen/core/PlayerCharacter.java [hunk 3]
```diff
@@ -1158,17 +1158,7 @@ public int getEquipSetNumber()
...
-		for (Equipment eq : aList)
-		{
-			if (aString.equalsIgnoreCase(eq.getName()))
-			{
-				match = eq;
-			}
-		}
-
-		return match;
+		return aList.stream().filter(v -> aString.equalsIgnoreCase(v.getName())).findFirst().get();
```
Comment (thpr) [inline code/src/java/pcgen/core/PlayerCharacter.java]: Logic change.  the FOR loop does LAST, the stream does first

File: code/src/java/pcgen/core/PlayerCharacter.java [hunk 4]
```diff
@@ -8105,14 +8098,7 @@ public void setTemplatesAdded(CDOMObject po, PCTemplate pct)
...
 	{
-		for (PCClass cl : getClassSet())
-		{
-			if (isClassSkill(cl, sk))
-			{
-				return true;
-			}
-		}
-		return false;
+		return getClassSet().parallelStream().anyMatch(v -> isClassSkill(v, sk));
```
Comment (thpr) [inline code/src/java/pcgen/core/PlayerCharacter.java]: parallel here too

Comment (grimreaper) [inline code/src/java/pcgen/core/PlayerCharacter.java]: fixed

**Pull request closed without merging**